### PR TITLE
Update cool model unit test

### PR DESCRIPTION
**Pull request by KimballCai**
Separate the CoolModelTest into different unit tests.

### CODE REVIEW

File: cool-core/src/main/java/com/nus/cool/core/iceberg/aggregator/CountAggregator.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public void process(Map<String, BitSet> groups, FieldRS field,
             }
             if(resultMap.get(entry.getKey()).getCount() == null) {
                 BitSet bs = entry.getValue();
-                int count = 0;
+                float count = 0;
```
Comment (hugy718) [inline cool-core/src/main/java/com/nus/cool/core/iceberg/aggregator/CountAggregator.java:45]: should be int?

File: cool-core/src/main/java/com/nus/cool/core/iceberg/aggregator/MaxAggregator.java [hunk 2]
```diff
@@ -42,7 +42,7 @@ public void process(Map<String, BitSet> groups, FieldRS field,
             }
             if (resultMap.get(entry.getKey()).getMax() == null) {
                 BitSet bs = entry.getValue();
-                int max = 0;
+                float max = 0;
```
Comment (hugy718) [inline cool-core/src/main/java/com/nus/cool/core/iceberg/aggregator/MaxAggregator.java:45]: InputVector::get returns int yet results in the resultMap are in float. May be we shall change them all to int?

File: cool-core/src/main/java/com/nus/cool/core/iceberg/aggregator/MinAggregator.java [hunk 3]
```diff
@@ -42,7 +42,7 @@ public void process(Map<String, BitSet> groups, FieldRS field,
             }
             if (resultMap.get(entry.getKey()).getMin() == null) {
                 BitSet bs = entry.getValue();
-               int min = Integer.MAX_VALUE;
+                float min = Integer.MAX_VALUE;
```
Comment (hugy718) [inline cool-core/src/main/java/com/nus/cool/core/iceberg/aggregator/MinAggregator.java:45]: same here

File: cool-core/src/main/java/com/nus/cool/model/CoolModel.java [hunk 4]
```diff
@@ -42,193 +42,190 @@
...
+
+        // Only load the latest version
+        File currentVersion = versions[versions.length - 1];
+
+        // Read schema information
+        TableSchema schema = TableSchema.read(new FileInputStream(new File(currentVersion, "table.yaml")));
+        CubeRS cubeRS = new CubeRS(schema);
+
+        File[] cubletFiles = currentVersion.listFiles((file, s) -> s.endsWith(".dz"));
+        // System.out.println("Cube " + cube + ", Use version: " + currentVersion.getName());
```
Comment (hugy718) [inline cool-core/src/main/java/com/nus/cool/model/CoolModel.java:118]: remove the commented out println statements.

File: cool-core/src/test/java/com/nus/cool/functionality/CsvLoaderTest.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                "java",
+                "com",
+                "nus",
+                "cool",
+                "core",
+                "resources").toString();
+        return new Object[][] {
+                {
+                        "health",
+                        // Paths.get(sourcePath, "health", "table.yaml").toString(),
```
Comment (hugy718) [inline cool-core/src/test/java/com/nus/cool/functionality/CsvLoaderTest.java:51]: commented lines

File: cool-core/src/test/java/com/nus/cool/functionality/CohortAnalysisTest.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                        out1
+                },{
+                        Paths.get(System.getProperty("user.dir"),  "..", "datasetSource").toString(),
+                        Paths.get(System.getProperty("user.dir"),  "..", "health", "query1-1.json").toString(),
+                        out2
+                }
+        };
+    }
+
+}
```
Comment (hugy718) [inline cool-core/src/test/java/com/nus/cool/functionality/CohortAnalysisTest.java:88]: insert newline

**Review by hugy718 — CHANGES_REQUESTED** [review]
let lingze whether the unit test are standard.

Besides, let's discuss the mismatch of int and float in iceberg module.

File: cool-core/src/test/java/com/nus/cool/functionality/CsvLoaderTest.java (hunk 5 above)
Comment (KimballCai) [reply cool-core/src/test/java/com/nus/cool/functionality/CsvLoaderTest.java:51]: these commented lines maybe used later

File: cool-core/src/main/java/com/nus/cool/core/iceberg/aggregator/MinAggregator.java (hunk 3 above)
Comment (KimballCai) [reply cool-core/src/main/java/com/nus/cool/core/iceberg/aggregator/MinAggregator.java]: update the min

File: cool-core/src/main/java/com/nus/cool/core/iceberg/aggregator/MaxAggregator.java (hunk 2 above)
Comment (KimballCai) [reply cool-core/src/main/java/com/nus/cool/core/iceberg/aggregator/MaxAggregator.java]: update the max

File: cool-core/src/main/java/com/nus/cool/core/iceberg/aggregator/CountAggregator.java (hunk 1 above)
Comment (KimballCai) [reply cool-core/src/main/java/com/nus/cool/core/iceberg/aggregator/CountAggregator.java]: update the count

File: cool-core/src/test/java/com/nus/cool/functionality/CohortAnalysisTest.java (hunk 6 above)
Comment (KimballCai) [reply cool-core/src/test/java/com/nus/cool/functionality/CohortAnalysisTest.java:88]: delete

File: cool-core/src/test/java/com/nus/cool/functionality/CohortSelectionTest.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        CoolModel coolModel = new CoolModel(datasetPath);
+        coolModel.reload(inputSource);
+
+        CubeRS cube = coolModel.getCube(query.getDataSource());
+
+        List<Integer> cohortResults = coolModel.cohortEngine.selectCohortUsers(cube,null, query);
+        List<Integer> selectionGlobalIDs = Arrays.asList(0, 2, 3, 4, 5, 7, 9, 11, 12);
+        assert cohortResults.equals(selectionGlobalIDs);
+        List<String> userIDs = coolModel.cohortEngine.listCohortUsers(cube, cohortResults);
+        List<String> selectionActualIDs = Arrays.asList("P-0", "P-2", "P-3", "P-4", "P-5", "P-7", "P-9", "P-11", "P-12");
```
Comment (Zrealshadow) [inline cool-core/src/test/java/com/nus/cool/functionality/CohortSelectionTest.java:49]: Pass in as a parameter, decoupling data and logic

File: cool-core/src/test/java/com/nus/cool/functionality/CsvLoaderTest.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    public void setUp() {
+        logger.info("Start UnitTest " + CsvLoaderTest.class.getSimpleName());
+    }
+
+    @AfterTest
+    public void tearDown() {
+        logger.info(String.format("Tear down UnitTest %s\n", CsvLoaderTest.class.getSimpleName()));
+    }
+
+    @Test(dataProvider = "CsvLoaderTestDP")
```
Comment (Zrealshadow) [inline cool-core/src/test/java/com/nus/cool/functionality/CsvLoaderTest.java:29]: For this method which you can not prove it's true, you can add wrong test and check the exception.

File: cool-core/src/test/java/com/nus/cool/functionality/FunnulAnalysisTest.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    public void FunnelAnalysisUnitTest(String datasetPath, String queryPath, int[] out) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        FunnelQuery query = mapper.readValue(new File(queryPath), FunnelQuery.class);
+
+        String inputSource = query.getDataSource();
+        CoolModel coolModel = new CoolModel(datasetPath);
+        coolModel.reload(inputSource);
+
+        if (!query.isValid())
+            throw new IOException("[x] Invalid cohort query.");
```
Comment (Zrealshadow) [inline cool-core/src/test/java/com/nus/cool/functionality/FunnulAnalysisTest.java:43]: Don't throw exceptions in Unittest

File: cool-core/src/test/java/com/nus/cool/functionality/IcebergLoaderTest.java [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+        // load .dz file
+        String dataSourceName = query.getDataSource();
+        CoolModel coolModel = new CoolModel(datasetPath);
+        coolModel.reload(dataSourceName);
+
+        // execute query
+        List<BaseResult> results = coolModel.olapEngine.performOlapQuery(coolModel.getCube(dataSourceName), query);
+
+        assert results.equals(out);
```
Comment (Zrealshadow) [inline cool-core/src/test/java/com/nus/cool/functionality/IcebergLoaderTest.java:47]: Do not use `assert` keyword in java, use the `Assert.assert` from TestNG.
`assert` keyword is rarely used in Java. JVM will ignore `assert` command by default. 
In compiling, only you add flag `-eableassertions`  and `assert` is actived

**Review by Zrealshadow — CHANGES_REQUESTED** [review]

File: cool-core/src/test/java/com/nus/cool/functionality/CohortSelectionTest.java (hunk 7 above)
Comment (KimballCai) [reply cool-core/src/test/java/com/nus/cool/functionality/CohortSelectionTest.java:49]: update

File: cool-core/src/test/java/com/nus/cool/functionality/FunnulAnalysisTest.java (hunk 9 above)
Comment (KimballCai) [reply cool-core/src/test/java/com/nus/cool/functionality/FunnulAnalysisTest.java:43]: update

File: cool-core/src/test/java/com/nus/cool/functionality/IcebergLoaderTest.java (hunk 10 above)
Comment (KimballCai) [reply cool-core/src/test/java/com/nus/cool/functionality/IcebergLoaderTest.java:47]: replaced

**Review by Zrealshadow — CHANGES_REQUESTED** [review]
Can not pass all unit test. 
Log:
> Failed tests: 
> CubeListUnitTest(com.nus.cool.core.model.CoolModelTest)

**Review by Zrealshadow — APPROVED** [review]

**Pull request merged**